### ISSUE TITLE
Two-factor auth CLI[Question]

### ISSUE THREAD

**Issue by chocochrispi** [caprover/caprover#1854]
How do we deploy with Two-factor Auth activated?

**Comment by Androz2091** [comment]
👍

**Comment by githubsaturn** [comment]
Very good question! When using 2FA it is a good practice to avoid using password based CLIs. For deployments, you should use app tokens.

If you really need to use the CLI with password, you can pass the 2FA token as an environment variable to any CapRover command, for example:
```bash
CAPROVER_OTP_TOKEN=123456; caprover login
```

**Comment by githubsaturn** [comment]
For app tokens, you can enable them on the apps' deployment dashboard and use them in `caprover deploy` command:
```
caprover deploy --appToken 123456123456 ..........
```

**Comment by drmrbrewer** [comment]
@githubsaturn so it's a different app token per app?  I have 73 apps so it would be quite onerous to generate and manage a whole bunch of different app tokens.  Do I understand correctly that, even though it is "good practice to avoid using password based CLIs", the intention is to continue to support this indefinitely?  And in `CAPROVER_OTP_TOKEN=123456; caprover login`, is `CAPROVER_OTP_TOKEN` the token you get when link a new server to the pro plan via [this page](https://pro.caprover.com/#/servers), i.e. one per instance?  My current thought is to enable 2FA to protect the web-based login page but continue to use the password for app deploys from the server... though it's been so long since I had to actually do a `caprover login` manually that I've forgotten whether you can `login` once and then `deploy` without having to supply the password via the cli.

**Comment by githubsaturn** [comment]
> it's a different app token per app? 

Yes, and that's the goal for these tokens: to limit the blast radius in case of leak. 


>  Do I understand correctly that, even though it is "good practice to avoid using password based CLIs", the intention is to continue to support this indefinitely


Yes, the support for the password based operations will continue to exist in the CLI commands. 

> is CAPROVER_OTP_TOKEN the token you get when link a new server to the pro plan via [this page](https://pro.caprover.com/#/servers), i.e. one per instance? 

Yes.
